packages/upload-api/src/blob/allocate.js

```diff
@@ -0,0 +1,106 @@
+import * as Server from '@ucanto/server'
+import * as W3sBlob from '@web3-storage/capabilities/web3.storage/blob'
+import * as API from '../types.js'
+import { ensureRateLimitAbove } from '../utils/rate-limits.js'
+
+/**
+ * @param {API.W3ServiceContext} context
+ * @returns {API.ServiceMethod<API.BlobAllocate, API.BlobAllocateSuccess, API.BlobAllocateFailure>}
+ */
+export function blobAllocateProvider(context) {
+  return Server.provide(
+    W3sBlob.allocate,
+    async ({ capability, invocation }) => {
+      const { blob, cause, space } = capability.nb
+      let size = blob.size
+
+      // Rate limiting validation
+      // TODO: we should not produce rate limit error but rather suspend / queue task to be run after enforcing a limit without erroring
+      const rateLimitResult = await ensureRateLimitAbove(
+        context.rateLimitsStorage,
+        [space],
+        0
+      )
+      if (rateLimitResult.error) {
```

Comment:
I find it very confusing that rateLimitResult provides some error but we throw it away and then say space is blocked. I'm not sure why do we conclude space is blocked or why do we discard error that rate limit check produced, should not that error be more accurate in describing what happened ?

Reply:
I feel we already talked about this. I am just keeping doing what `store` does and not trying to also figure out what this rate limit should do. What is your suggestion? I am honestly feeling it is better to just remove this for now

Reply:
I quite adding the rate limit thing. We won't be able to block users (which I think is the reason for this) until we figure this out, but I don't want to block on this